src/main/java/demo/AclCheckingEntityListener.java

```diff
@@ -46,11 +52,31 @@
 public class AclCheckingEntityListener {
 
 	// TODO: Use proper dependency injection
-	private AclQueryAugmentor<Object> augmentor = new AclQueryAugmentor<Object>();
+	private AclJpaQueryAugmentor<Object> augmentor = new AclJpaQueryAugmentor<Object>();
+
+	private MutableAclService mutableAclService;
+
+	private PermissionFactory permissionFactory;
 
 	// TODO: Use proper dependency injection
 	public static JpaContext context;
 
```

Comment:
@rwinch any ideas on how to do proper dependency injection here? Or should we consider only protecting repository invocations?